tests/unit/youtube/factory-tests.js

```diff
@@ -0,0 +1,135 @@
+/*
+ * https://github.com/Automattic/expect.js
+ */
+
+describe('Youtube factory tests', function () {
+  // Load test helpers
+  require('../../utils/test_lib.js');
+
+  // Setup
+  var pluginFactory;
+  OO.Video = { plugin: function(plugin) { pluginFactory = plugin; } };
+
+  // override video element canPlayType to return true always
+  var oldCreateElement = document.createElement;
+  document.createElement = _.bind(function(type) {
+    if (type === "div") {
+      return { canPlayType: function() { return true; }};
+    } else {
+      return oldCreateElement(type);
+    }
+  }, this);
+
+  // Load file under test
+  jest.dontMock('../../../src/youtube/js/youtube');
+  require('../../../src/youtube/js/youtube');
+
+  // restore document.createElement
+  document.createElement = oldCreateElement;
+
+  var vtc = { EVENTS: { CAN_PLAY: "can_play" },
+                        notify: function(){} };
+
+
+
+  it('should contain parameter \'name\'', function () {
+    expect(pluginFactory.name).to.be.ok();
+  });
+
+  it('should provide a list of supported encodings', function(){
+    // This is controlled by document.createElement("video").canPlayType(type);
```

Comment:
Is this comment true? OO.VIDEO.ENCODING.MP4 is set in encodings without regard to document.createElement('video') (or at least it seems that way). Just wanted to double check.